.github/workflows/cli.yml

```diff
@@ -73,265 +45,135 @@ jobs:
-        with:
-          profile: minimal
-          toolchain: ${{ env.RUST_VERSION }}
-          target: x86_64-apple-darwin
-          override: false
-
-      - name: install Rust stable
-        uses: actions-rs/toolchain@v1
-        # FIXME: We use rust-toolchain.toml to specify targets, etc. Unfortunately,
-        # the action doesn't support it yet (https://github.com/actions-rs/toolchain/issues/208).
-        # Once it does, we should be able to remove the `with` block entirely.
+      - name: Rust job init
+        uses: ./.github/actions/init_rust_job
         with:
-          profile: minimal
-          toolchain: ${{ env.RUST_VERSION }}
-          target: aarch64-apple-darwin
-          override: false
+          cdn: ${{ github.ref_name == 'main' && env.PROD_ASSETS_BUCKET || env.DEV_ASSETS_BUCKET }}
+          assets_version: ${{ env.ASSETS_VERSION }}
 
-      - name: set up a cache for .cargo
-        uses: actions/cache@v3
-        timeout-minutes: 5
-        continue-on-error: true
-        with:
-          path: ~/.cargo/*
-          key: ${{ runner.os }}-${{ runner.arch }}-${{ hashFiles('Cargo.lock', 'Cargo.toml') }}
-
-      - name: set up a cache for target
-        uses: actions/cache@v3
-        timeout-minutes: 5
-        continue-on-error: true
-        with:
-          path: target
-          key: ${{ runner.os }}-${{ runner.arch }}-${{ hashFiles('Cargo.lock', 'Cargo.toml') }}
-
-      - name: fetch assets
-        run: |
-          curl https://temp-artifact-storage.s3.eu-west-3.amazonaws.com/assets.zip --output assets.zip
-          unzip assets.zip
-          rm -r crates/server/assets
-          mv assets crates/server/assets
-          cp -r crates/server/assets ~/.grafbase/
-          touch ~/.grafbase/version.txt
-
-      - name: test
+      - name: Test
         uses: actions-rs/cargo@v1
         with:
           command: test
 
-      - name: build
+      - name: Build aarch64 release
         uses: actions-rs/cargo@v1
-        if: github.ref == 'refs/heads/main'
+        if: startsWith(github.ref, 'refs/tags/')
         with:
           use-cross: true
           command: build
           args: "--release --target aarch64-apple-darwin"
 
-      - name: build
+      - name: Build x86_64 release
         uses: actions-rs/cargo@v1
-        if: github.ref == 'refs/heads/main'
+        if: startsWith(github.ref, 'refs/tags/')
         with:
           use-cross: true
           command: build
           args: "--release --target x86_64-apple-darwin"
 
-      - name: chmod binary
-        if: github.ref == 'refs/heads/main'
-        run: chmod +x target/x86_64-apple-darwin/release/grafbase
-
-      - name: chmod binary
-        if: github.ref == 'refs/heads/main'
-        run: chmod +x target/aarch64-apple-darwin/release/grafbase
+      - name: chmod binaries
+        if: startsWith(github.ref, 'refs/tags/')
+        shell: bash
+        run: |
+          chmod +x target/x86_64-apple-darwin/release/grafbase
```

Comment:
IIRC this didn't actually work and I needed to manually do it again, might be related to downloads not being executable though

Reply:
Idd, I kept it because it was in place but I'm keen on removing. Usually we have to add the permissions to allow execution

Reply:
It definitely needs to be done before release, I'm just not sure it should be done here.
If it's not executable it doesn't work when distributing via NPM

Reply:
What's `Idd`?